pkg/operator/controller/gatewayapi/controller.go

```diff
@@ -25,7 +27,10 @@ import (
 )
 
 const (
-	controllerName = "gatewayapi_controller"
+	controllerName                    = "gatewayapi_controller"
+	experimantalGatewayAPIGroupName   = "gateway.networking.x-k8s.io"
+	crdAPIGroupIndexFieldName         = "crdAPIGroup"
+	gatewayCRDAPIGroupIndexFieldValue = "gateway"
```

Comment:
A one-word indexer feels too expansive.  Why not at least "gateway.networking"?

Reply:
The controller-runtime documentation actually says to use the field name (in our case, that would be `"spec.group"`), and the index value should be the field value (for example, `"gateway.networking.x-k8s.io"`).  But we have written a lot of code that doesn't follow the prescribed idiom, and it seems to work fine, so I suspect that the index key can be any string that is unique among indexes within the manager.  



https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#FieldIndexer

Reply:
Make sense to only be looking at the spec.group.